nsxt/resource_nsxt_policy_ipsec_vpn_session.go

```diff
@@ -0,0 +1,630 @@
+
+	_, err1 := resourceNsxtPolicyIPSecVpnSessionExists(gwID, localeServiceID, serviceID, id, connector)
+	if err1 == nil {
+		return fmt.Errorf("IPSecVpnSession with nsx_id '%s' already exists under IPSecVpnService '%s' of localeService '%s'", id, serviceID, localeServiceID)
+	} else if !isNotFoundError(err1) {
+		return err1
+	}
+
+	obj, err2 := getIPSecVPNSessionFromSchema(d)
+	if err2 != nil {
+		return err2
+	}
+
+	client := ipsec_vpn_services.NewSessionsClient(connector)
+
+	err3 := client.Patch(gwID, localeServiceID, serviceID, id, obj)
+
+	if err3 != nil {
+		return handleCreateError("IPSecVpnSession", id, err3)
+	}
+
+	d.SetId(id)
+	d.Set("nsx_id", id)
+
+	return resourceNsxtPolicyIPSecVpnSessionRead(d, m)
+}
+
+func resourceNsxtPolicyIPSecVpnSessionExists(tier0ID string, localeServiceID string, serviceID string, sessionID string, connector *client.RestConnector) (bool, error) {
+	client := ipsec_vpn_services.NewSessionsClient(connector)
+	_, err := client.Get(tier0ID, localeServiceID, serviceID, sessionID)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, err
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func resourceNsxtPolicyIPSecVpnSessionRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	converter := bindings.NewTypeConverter()
+	converter.SetMode(bindings.REST)
+
+	id := d.Id()
+	resourceType := d.Get("vpn_type").(string)
+	if id == "" {
+		return fmt.Errorf("Error obtaining IPSecVpnSession ID")
+	}
+
+	servicePath := d.Get("service_path").(string)
+	isT0, gwID, localeServiceID, serviceID, err := parseIPSecVPNServicePolicyPath(servicePath)
+	if err != nil {
+		return err
+	}
+	if !isT0 {
+		return fmt.Errorf("VPN is supported only on Tier-0 with ACTIVE-STANDBY HA mode.")
+	}
+
+	client := ipsec_vpn_services.NewSessionsClient(connector)
+
+	obj, err1 := client.Get(gwID, localeServiceID, serviceID, id)
+
+	if err1 != nil {
+		if isNotFoundError(err1) {
+			d.SetId("")
+			return nil
+		}
+		return handleReadError(d, "VPN Session", id, err1)
+	}
+
+	if resourceType == model.IPSecVpnSession_RESOURCE_TYPE_ROUTEBASEDIPSECVPNSESSION {
+		interfaceVpn, errs := converter.ConvertToGolang(obj, model.RouteBasedIPSecVpnSessionBindingType())
+		if len(errs) > 0 {
+			return fmt.Errorf("Error converting VPN Session %s", errs[0])
+		}
+		blockVPN := interfaceVpn.(model.RouteBasedIPSecVpnSession)
+
+		d.Set("display_name", blockVPN.DisplayName)
+		d.Set("description", blockVPN.Description)
+		setPolicyTagsInSchema(d, blockVPN.Tags)
+		d.Set("nsx_id", blockVPN.Id)
+		d.Set("path", blockVPN.Path)
+		d.Set("revision", blockVPN.Revision)
+		d.Set("vpn_type", blockVPN.ResourceType)
+		d.Set("authentication_mode", blockVPN.AuthenticationMode)
+		d.Set("compliance_suite", blockVPN.ComplianceSuite)
+		d.Set("connection_initiation_mode", blockVPN.ConnectionInitiationMode)
+		d.Set("enabled", blockVPN.Enabled)
+		if blockVPN.IkeProfilePath != nil {
```

Comment:
`d.Set` can handle nil value, so its safe to remove the check

Reply:
Done